x-pack/solutions/security/plugins/security_solution/public/timelines/components/modal/actions/open_timeline_button.test.tsx

```diff
@@ -20,9 +20,10 @@ jest.mock('../../../../sourcerer/containers');
 jest.mock('../../open_timeline/use_timeline_status');
 jest.mock('react-redux', () => {
   const origin = jest.requireActual('react-redux');
+  const mockDispatch = jest.fn();
   return {
     ...origin,
-    useDispatch: jest.fn(),
+    useDispatch: jest.fn(() => mockDispatch),
```

Comment:
This fixes 

- [x] [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/267663#01946e9c-d646-4bf0-b7d1-e89178a15545) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/267663/jobs/01946e9c-d646-4bf0-b7d1-e89178a15545/artifacts/01946ebe-0ca6-411a-b1a0-b7cf20082ade) Jest Tests / OpenTimelineButton should open the modal after clicking on the button 


The problem was that with React@18 the test started getting to [this](https://github.com/elastic/kibana/blob/da25d13a2ac5be97216e202bf5d6bda6ff920b3b/x-pack/solutions/security/plugins/security_solution/public/timelines/containers/all/index.tsx#L207) dispatch call and it failed because instead of a function we mock returned undefined.